tests/PactNet.Native.Tests/NativePactBuilderTests.cs

```diff
@@ -39,23 +56,147 @@ public void UponReceiving_WhenCalled_CreatesNewInteraction()
         }
 
         [Fact]
-        public void Build_WhenCalled_StartsMockServer()
+        public void Verify_WhenCalled_StartsMockServer()
         {
-            this.builder.Build();
+            this.builder.Verify(Success);
 
             this.mockServer.Verify(s => s.CreateMockServerForPact(this.handle, "127.0.0.1:0", false));
         }
 
         [Fact]
-        public void Build_ErrorStartingMockServer_ThrowsInvalidOperationException()
+        public void Verify_ErrorStartingMockServer_ThrowsInvalidOperationException()
         {
-            this.mockServer.Setup(
-                    s => s.CreateMockServerForPact(It.IsAny<PactHandle>(), It.IsAny<string>(), It.IsAny<bool>()))
+            this.mockServer
+                .Setup(s => s.CreateMockServerForPact(It.IsAny<PactHandle>(), It.IsAny<string>(), It.IsAny<bool>()))
                 .Throws<InvalidOperationException>();
 
-            Action action = () => this.builder.Build();
+            Action action = () => this.builder.Verify(Success);
 
             action.Should().Throw<InvalidOperationException>("because the mock server failed to start");
         }
+
+        [Fact(Skip = "Enable this when the released FFI library supports getting mock server logs")]
+        public void Verify_NoLogs_DoesNotWriteLogs()
+        {
+            this.builder.Verify(Success);
+
+            this.mockOutput.Verify(o => o.WriteLine(It.IsAny<string>()), Times.Never);
+        }
+
+        [Fact(Skip = "Enable this when the released FFI library supports getting mock server logs")]
+        public void Verify_Logs_WritesLogsToOutput()
+        {
+            this.mockServer.Setup(s => s.MockServerLogs(this.serverUri.Port)).Returns("some logs");
+
+            this.builder.Verify(Success);
+
+            this.mockOutput.Verify(o => o.WriteLine("some logs"));
+        }
+
+        [Fact]
+        public void Verify_InteractionThrowsException_RethrowsInteractionException()
+        {
+            this.builder
+                .Invoking(b => b.Verify(Error))
+                .Should().ThrowExactly<Exception>("because the inner interaction call failed");
+        }
+
+        [Fact]
+        public void Verify_NoMismatches_WritesPactFile()
+        {
+            this.builder.Verify(Success);
+
+            this.mockServer.Verify(s => s.WritePactFile(this.serverUri.Port, this.config.PactDir, false));
+        }
+
+        [Fact]
+        public void Verify_NoMismatches_ShutsDownMockServer()
+        {
+            this.builder.Verify(Success);
+
+            this.mockServer.Verify(s => s.CleanupMockServer(this.serverUri.Port));
+        }
+
+        [Fact]
+        public void Verify_FailedToWritePactFile_ThrowsInvalidOperationException()
```

Comment:
Not sure this test is necessary. You are basically just checking there is no try catch that are absorbing the underlying exception...  If in the future, you want to encapsulate the exception into a more functional one, this test will break.

Reply:
I agree - on first skim read, I thought this test was confirming that if the pact failed to write it would throw `InvalidOperationException`, but if I'm reading it correctly, it's just a mock that is throwing that exception?

Reply:
Oh, wait - is this here to make sure the exceptions are no longer being swallowed? As in, would this test have failed before this change?

Reply:
That's exactly the case. It's not the mock of the method called but the underlying method. I see why this test is here, but not 100% sure it's necessary to test it when writing the pact.

Reply:
Yes, this test proves that if the "inner" client code throws an exception then we don't swallow it. Without this, we could catch that exception and not rethrow it, which was the problem that this change is addressing